src/dbsync/src/sqlite/sqlite_dbengine.cpp

```diff
@@ -254,6 +254,69 @@ void SQLiteDBEngine::returnRowsMarkedForDelete(const nlohmann::json& tableNames,
     }
 }
 
+void SQLiteDBEngine::selectData(const std::string& table,
+                                const nlohmann::json& query,
+                                const DbSync::ResultCallback& callback)
+{
+    const auto& sql{ buildSelectQuery(table, query) };
+    auto const& stmt { getStatement(sql) };
+    const auto tableFields { m_tableFields[table] };
+
+    while (SQLITE_ROW == stmt->step())
+    {
+        Row row;
+        const auto& columns{ query.at("column_list") };
+        if (std::find_if(columns.begin(), columns.end(),[](const std::string& value){return value == "*";}) != columns.end())
+        {
```

Comment:
Please, add a brief comment saying why we need to do this if/else (* vs specific columns)